ui/src/app/views/workspace-list/component.html

```diff
@@ -36,33 +36,93 @@ <h3 class="table-header-descriptor">Your Workspaces</h3>
       <div class="create-workspace-text">Create a New Workspace</div>
       <clr-icon class="plus-circle" shape="plus-circle"></clr-icon>
     </div>
-    <div class="card workspace-card" *ngFor="let workspaceResponse of workspaceList" routerLink="{{workspaceResponse.workspace.namespace}}/{{workspaceResponse.workspace.id}}">
-      <h5 class="workspace-name">{{workspaceResponse.workspace.name}}</h5>
-      <div class="description">{{workspaceResponse.workspace.description}}</div>
+    <div class="card workspace-card" *ngFor="let workspacePermissions of workspaceList">
+      <div class="header-area">
+        <clr-dropdown [clrCloseMenuOnItemClick]="false">
+          <button type="button" [disabled]="actionsDisabled" clrDropdownTrigger>
+            <clr-icon shape="ellipsis-vertical" class="ellipsis-vertical" [class.disabled]="actionsDisabled"></clr-icon>
+          </button>
+          <clr-dropdown-menu *clrIfOpen>
+            <button class="action-item clone-item"
+                    routerLink="/workspaces/{{workspacePermissions.workspace.namespace}}/{{workspacePermissions.workspace.id}}/clone"
+                    routerLinkActive="active">
+              <clr-icon shape="copy"></clr-icon>
+              Clone
+            </button>
+            <span role="tooltip" aria-haspopup="true" class="tooltip tooltip-sm tooltip-top-right">
+              <button class="action-item edit-item"
+                      [disabled]="!workspacePermissions.canWrite"
+                      [class.disabled]="!workspacePermissions.canWrite"
```

Comment:
Your Angular-foo is becoming quite formidable.

Reply:
This was actually Greg!